tests/foreman/api/test_classparameters.py

```diff
@@ -15,78 +15,188 @@
+        },
+        {
+            u'sc_type': 'integer',
+            u'value': gen_string('utf8'),
+        },
+        {
+            u'sc_type': 'real',
+            u'value': gen_string('alpha'),
+        },
+        {
+            u'sc_type': 'array',
+            u'value': '0',
+        },
+        {
+            u'sc_type': 'hash',
+            u'value': 'a:test',
+        },
+        {
+            u'sc_type': 'yaml',
+            u'value': '{a:test}',
+        },
+        {
+            u'sc_type': 'json',
+            u'value': gen_string('alpha'),
+        },
+    ]
+
+
 class SmartClassParametersTestCase(APITestCase):
     """Implements Smart Class Parameter tests in API"""
 
+    @classmethod
+    def setUpClass(cls):
+        super(SmartClassParametersTestCase, cls).setUpClass()
+        cls.host_name = settings.server.hostname
+        cls.host = entities.Host().search(
+            query={'search': 'name=' + cls.host_name}
+        )[0]
+        ssh.command('puppet module install --force puppetlabs/ntp')
+        cls.env = entities.Environment().search(
+            query={'search': 'name="production"'}
+        )[0]
+        cls.proxy = entities.SmartProxy(name=cls.host_name).search()[0]
+        cls.proxy.import_puppetclasses(environment=cls.env)
+        cls.puppet = entities.PuppetClass().search(
+            query={'search': 'name="ntp"'}
+        )[0]
+        cls.sc_params_list = entities.SmartClassParameters().search(
+            query={'search': 'puppetclass="ntp"', 'per_page': 100}
+        )
+        if len(cls.sc_params_list) < 45:
+            raise RuntimeError('There are not enough smart class parameters to'
+                               ' work with in provided puppet class')
+
     @run_only_on('sat')
-    @stubbed()
-    @tier3
+    @tier2
     def test_positive_list_parameters_by_host_id(self):
         """List all the parameters included in specific Host by its id.
 
         @id: a9e551f9-261b-40e6-b7f6-35621fc46285
 
         @assert: Parameters listed for specific Host.
 
-        @caseautomation: notautomated
-
-        @CaseLevel: System
+        @CaseLevel: Integration
         """
+        sc_param = self.sc_params_list.pop()
+        sc_param.override = True
+        sc_param.update(['override'])
+        self.assertEqual(sc_param.read().override, True)
+        self.host.puppet_class = [self.puppet]
+        self.host.update(['puppet_class'])
+        self.assertGreater(len(self.host.list_scparams()), 0)
 
     @run_only_on('sat')
-    @stubbed()
-    @tier3
+    @tier2
     def test_positive_list_parameters_by_hostgroup_id(self):
         """List all the parameters included in specific HostGroup by id.
 
         @id: 88ceea89-b8b5-4ca2-9d59-3b2614c7f9a7
 
         @assert: Parameters listed for specific HostGroup.
 
-        @caseautomation: notautomated
-
-        @CaseLevel: System
+        @CaseLevel: Integration
         """
+        sc_param = self.sc_params_list.pop()
```

Comment:
For example, we have one puppet module in that case `ntp` that contains around 50 puppet classes. After we import it to the Sat application we can perform interaction with them. We need to use unique puppet class for each test case as each interaction can possibly impact any other interaction from another test case, so we need to choose exactly unaffected puppet class, so I am taking last element from the list and remove it from that list to guarantee that we never take used puppet class for test case. Of course, we can choose random element, but anyway we need to remove it from the list.
It was an idea to remove puppet module or puppet environment, but after we re-import it, system remember old one values and adding a new one, so we have a chance to met same puppet class for interaction